src/Classes/DKP_Entry.lua

```diff
@@ -143,50 +143,42 @@ function DKP_Entry:GetFormattedNames()
         end
     end
 
-    local formattedNames = ''
-
-    local function compare(a,b)
-        local a_member = Guild:GetMemberByName(a)
-        local b_member = Guild:GetMemberByName(b)
-
-        if a_member == nil then
-            return true
-        elseif b_member == nil then
-            return false
-        end
-
-        if a_member == nil or b_member == nil then
-            print('ERROR sorting entry names: ', a, b)
-            print(a_member.name, b_member.name)
-            return false
+    local memberTable = {}
+    for key, name in ipairs(self.names) do
+        local member = Guild:GetMemberByName(name)
+        if member ~= nil then
+            table.insert(memberTable, member)
         end
+    end
 
-        if b_member.class == a_member.class then
-            return b_member.name > a_member.name
+    local function compare(a,b)
+        if b.class == a.class then
+            return b.name > a.name
         else
-            return b_member.class > a_member.class
+            return b.class > a.class
         end
-        return false
     end
-    --pcall(table.sort(self.names, compare))
-    table.sort(self.names, compare)
 
-    local remove_names = {};
-    for key, name in pairs(self.names) do
-        local member = Guild:GetMemberByName(name)
+    --pcall(table.sort(self.names, compare))
+    table.sort(memberTable, compare)
 
-        if member ~= nil then
-            formattedNames = formattedNames .. member.formattedName
-            if key ~= #self.names then formattedNames = formattedNames .. ', ' end
-        else
-            remove_names[key] = name
-        end
+    local formattedNames = ''
+    for key, member in pairs(memberTable) do
+        if key ~= 1 then formattedNames = formattedNames .. ', ' end
+        formattedNames = formattedNames .. member.formattedName
```

Comment:
There was an issue here where if the last person in the list was not a guild member, formattedNames would end in a comma. I changed the comma to prepend on all but the first item to fix this case.